ocaml/xapi/xapi_cluster.ml

```diff
@@ -88,6 +89,7 @@ let destroy ~__context ~self =
       Db.Cluster_host.destroy ~__context ~self:ch
     ) cluster_host;
     Db.Cluster.destroy ~__context ~self;
+    set_ha_cluster_stack_if_ha_enabled ~__context ~value:"xhad";
```

Comment:
We should not hardcode "xhad" here. See Cluster_stack_constraints.choose_cluster_stack.
Ideally you would just call that to tell you the cluster stack, and then all you need is an update_ha_cluster_stack.